src/euphorie/client/notifications/base.py

```diff
@@ -43,12 +47,24 @@ def recipient(self):
     def index(self):
         """The mail text."""
         full_name = self.webhelpers.get_user_fullname(self.account)
-        session_links = "\n".join(
-            [
-                f"* [{session.title}]({session.absolute_url()}/@@start)"
-                for session in self.sessions
-            ]
-        )
+
+        session_links = ""
+        for session in self.sessions:
+            session_url = ""
+            try:
+                session_url = session.absolute_url()
+            except ValueError:
+                logger.warn(
+                    "Could not get session URL for session %s and "
+                    "zodb_path %s (session id: %s). There might be a data "
+                    "inconsistency. Not including this session in the "
+                    "notification mailing.",
+                    session.title,
```

Comment:
Please do not use `%s`  when substituting what could be a multiline text, use `%r` we were already been asked to do something similar.

Reply:
Done. would the title for example be multiline? and how would `__repr__` over `__str__` make a difference?
However, I have changed that.

Reply:
okay, there's a difference in escaping:

```
>>> "ab %s c" % """okay
... nay"""
'ab okay\nnay c'

>>> "ab %r c" % """okay
... nay"""
"ab 'okay\\nnay' c"
```